### PR TITLE
New web 2024

### PR DESCRIPTION
**Pull request by WilloS88**

### CODE REVIEW

File: src/components/countdown/Countdown.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft())
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setTimeLeft(calculateTimeLeft())
+    }, 1000)
+
+    return () => clearTimeout(timer)
+  }, [timeLeft])
```
Comment (Rajce23) [inline src/components/countdown/Countdown.tsx:31]: do it better

**Review by FilipDlouhy — DISMISSED** [review]

File: public/hero/HeroBackground.png
Comment (tomesfilip) [inline public/hero/HeroBackground.png:1]: is this okay? seems like an empty image

File: index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
```
Comment (tomesfilip) [inline index.html]: change to cs lang, also check the correct value for the czech language attribute

File: src/components/countdown/Countdown.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  return (
+    <div className="flex items-center justify-center space-x-1">
+      <div className="flex">
```
Comment (tomesfilip) [inline src/components/countdown/Countdown.tsx:35]: these outer divs seem to be unnecessary, maybe we can get rid of them?

File: src/components/sections/AboutSection.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import Socha from "../../../public/Stani2.png"
+import InfoCard from "../InfoCard"
+
+const AboutSection = () => {
+  return (
+    <div className="mt-10 flex justify-center items-center">
+      <div className="grid grid-cols-2 grid-rows-6">
+        <div className="row-start-1 row-end-4 col-start-1 col-end-2">
+          <img src={Socha} alt="" />
```
Comment (tomesfilip) [inline src/components/sections/AboutSection.tsx]: when using the public path, we can put it directly to the `src="/Stani2.png"`
- also, provide the alt value

File: src/components/sections/AboutSection.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+              Nemusíte se bát ani kožešin či rukaviček. Nezapomeňte na podpadky,
+              které dodají vašemu outfitu mrazivou tečku.
+            </p>
+          </InfoCard>
+        </div>
+        <div className="row-start-4 row-end-7 col-start-2 col-end-3">
+          <InfoCard headline="PROGRAM:">
+            <div className="space-y-2">
+              <p>
+                <strong>19:00</strong> - kokot1
```
Comment (tomesfilip) [inline src/components/sections/AboutSection.tsx]: I think, we don't want these here :D

File: src/components/sections/AboutSection.tsx
Comment (tomesfilip) [inline src/components/sections/AboutSection.tsx:1]: when there's a repeating component multiple times, consider creating a reusable component that will just accept the corresponding data. (DRY - don't repeat yourself principle)

File: src/components/sections/HeroSection.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import Heading from "../../../public/hero/HeroHeading.svg"
+import Clouds from "../../../public/hero/HeroClouds.png"
+import Pillars from "../../../public/hero/HeroPillarsCropped.png"
```
Comment (tomesfilip) [inline src/components/sections/HeroSection.tsx]: the same with images, when using public ones, just use them directly in the component

File: src/components/sections/HeroSection.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+import Countdown from "../countdown/Countdown"
+
+const HeroSection = () => {
+  const gradientStyle = {
+    backgroundImage: `linear-gradient(to top, rgba(194,194,194,0.32816876750700286) 0%, rgba(255,255,255,1) 32%)`,
```
Comment (tomesfilip) [inline src/components/sections/HeroSection.tsx]: no need for the special ticks, use normal ''

File: src/components/sections/HeroSection.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <div
+      style={gradientStyle}
+      className="flex flex-col justify-center items-center "
+    >
+      <div className="mt-16 mx-5">
+        <img src={Heading} alt="Heading" className="" />
+      </div>
+      <div className="my-5">
+        <Countdown targetDate={new Date("2024-02-16T19:00:00")} />
```
Comment (tomesfilip) [inline src/components/sections/HeroSection.tsx]: save the date into a constant somewhere

File: src/components/sections/HeroSection.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    backgroundRepeat: "no-repeat",
+  }
+
+  return (
+    <div
+      style={gradientStyle}
+      className="flex flex-col justify-center items-center "
+    >
+      <div className="mt-16 mx-5">
+        <img src={Heading} alt="Heading" className="" />
```
Comment (tomesfilip) [inline src/components/sections/HeroSection.tsx]: we can remove the className as we're not using it here anyway

**Review by tomesfilip — CHANGES_REQUESTED** [review]
1. let's use type instead of interface, it's a common practice now, it has many advantages like inheritance, partial prop selection, and many more
2. for the component definition, let's use this pattern

type Props = {
 headline: string;
 children: ReactNode;
}
export const InfoCard = ({ headline, children }: Props) => {...}

File: src/components/countdown/Countdown.tsx (hunk 3 above)
Comment (WilloS88) [reply src/components/countdown/Countdown.tsx:35]: Im afraid it is necessary

**Review by tomesfilip — APPROVED** [review]

**Pull request merged**